enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+early version of BuildController mirrors this strategy. Within the build
+context, this ConfigMap is gunzipped and base64-decoded before being fed to
+Ignition live-apply.
+
+This currently looks like this:
+
+```mermaid
+flowchart TB
+  Start("Start")
+  subgraph Prepare [Prepare For Build]
+    direction TB
+    RenderContainerfile("Render Containerfile")
+    PrepMachineConfig("gzip and base64 \n rendered MachineConfig")
+    ConfigMaps("Create ephemeral ConfigMaps")
+  
+    RenderContainerfile --> ConfigMaps
+    PrepMachineConfig --> ConfigMaps
+  end
+  
+  subgraph ImageBuilder [Image Builder Context]
+    direction TB
+    DecodeAndExtract("Decode and extract Ignition \n from MachineConfig ConfigMap")
+    IgnitionLiveApply("$ ignition live-apply")
+    RunAdminProvidedContent("Run admin-provided \n Containerfile content")
+  
+    DecodeAndExtract-->IgnitionLiveApply-->RunAdminProvidedContent
+  end
+
+  Stop("Stop")
+
+  Start --> Prepare
+  ConfigMaps --> |Mounted as volumes into image build context| ImageBuilder
+  IsSuccess{"Is Build Successful?"}
+
+  ImageBuilder --> IsSuccess
+  IsSuccess --> | Yes | DeleteConfigMaps("Delete ConfigMaps")
+  IsSuccess --> | No | LeaveConfigMaps("Leave ConfigMaps \n for inspection")
+  DeleteConfigMaps --> Stop
+  LeaveConfigMaps --> Stop
+```
+
+An alternative to this approach is to use a more sophisticated entrypoint
+for the build process that has permission to extract the needed information
+from the Kube API server and can render the Containerfile on-the-fly before
+beginning the build. A benefit of this approach is that cleanup would be
+fairly straightforward: Since everything would exist in a tempdir within the
+build pod, deletion of the build pod would perform the cleanups. The
+downside of this approach is that should a failure occur, fetching the build
+inputs for inspection and debugging would be more difficult.
+
+Once the bootc ConfigMap annotation standards are better known, we could
+potentially skip the gzip / base64-encoding step by converting MachineConfigs
+into multiple ConfigMaps, temporarily storing them in the API server, injecting
+them into the build context, then deleting them once the build is complete.
+
+Within the build context itself, we'd need a different mechanism than Ignition
+live-apply to consume those ConfigMaps and inject their contents into the
+appropriate places within the image being built. Additionally, one of the big
+ideas around the bootc ConfigMap spec is using annotations to indicate what
+directory to store the files in, what file modes they should have, etc. This
+data would be unavailable via a volume mount. So it may be necessary to create a
+map of ConfigMap name -> bootc annotations and store this in an additional
+ConfigMap. This would also necessitate the need for a binary such as Ignition
+live-apply to consume this information from within the build context to write
+those files into the appropriate places.
```

Comment:
This sounds very awkward. Would having a structured API for this make this easier to consume vs annotations?